### PR TITLE
Avoid reversing list just for the purpose of iteration.

### PR DESCRIPTION
**Pull request by paulomorgado**

### CODE REVIEW

File: src/Servers/HttpSys/src/FeatureContext.cs
Comment (Tratcher) [inline src/Servers/HttpSys/src/FeatureContext.cs:623]:
```suggestion
            for (var i = actions.Count - 1; i >= 0; i--)
```

Comment (paulomorgado) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: If you care that much about it, I have no problem.

Comment (Tratcher) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: The code as written doesn't work does it? It's failing a bunch of tests:
https://dev.azure.com/dnceng/public/_build/results?buildId=908050&view=ms.vss-test-web.build-test-results-tab&runId=28956888&resultId=123998&paneView=debug

Comment (paulomorgado) [reply src/Servers/HttpSys/src/FeatureContext.cs]: OOPS! Typo!

File: src/Servers/HttpSys/src/FeatureContext.cs
Comment (gfoidl) [inline src/Servers/HttpSys/src/FeatureContext.cs:623]:
```suggestion
            for (var i = actions.Count; i > 0; i--)
```
Is the "typical" way to write loops in C#.

But I don't care (just that it's said 😉).

Comment (paulomorgado) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: "typical" is a very subjective term.

That's a typical for loop for me.

Up until variables can be introduced in expressions. Then, it will be something like this:

```C#
while (var i = actions.Count; i-- > 0)
{
    // ..
}
```

Comment (gfoidl) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: I'm pretty sure that every for-loop within dotnet-org (and beyond) is written in the way as given in the [language reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/for). So we should stick to that convention.

Personally I find it harder to read in the form `for (var i = n; i-- > 0; )` as year long training for the eyes match `for (var i = n; i > 0; --i)` quite better.

Besides all that, the for-loop as written has bug (index violation) which is the most important thing to watch for.

Comment (paulomorgado) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: Check the examples at https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/for#examples

This use case is not there, but it doesn't make it less valid.

There's no `-->` in C#.

The expression is: `(i--) > 0`

Comment (gfoidl) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: > There's no `-->` in C#.

I know 😉 In some other languages the motivation to write loops like this, comes from the look-a-like of the variable going down to 0, so `--> 0` (analogy to limit).

But again, this is unusual in C# and especially in the dotnet-org and we should stay consistent.

For this point I've said enough and won't respond for this.

> Where is the bug?

In my brain...see other comment.

Comment (paulomorgado) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: Again, it's a style preference and "unusual" is in the eye of the beholder.

I'm OK with accepting a style that is equivalent in function and performance. I won't accept that there's anything wrong or unusual with the style I used.

Comment (En3Tho) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: In this case "beholder" is a whole team responsible to keep code not necessarily the best looking ever but consistent. There is nothing wrong with the you wrote your loop It's a valid code with no bugs. But it's just inconsistent with other code in this repo. And keeping thing consistent in a large code base is what helps people get through different pieces of code without trying to understand it's code conventions anew every time

Comment (paulomorgado) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: That's fair, and I already accepted that more than once.

Let's wait for someone that can to make the final decision.

Comment (BrennanConroy) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: > Let's wait for someone that can to make the final decision.

As maintainer of this repo I am asking you to please change it

Comment (paulomorgado) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: Sure! Can I have a reason for that?

Comment (JamesNK) [reply src/Servers/HttpSys/src/FeatureContext.cs:623]: To be consistent with other for loops in the repo.

Comment (paulomorgado) [reply src/Servers/HttpSys/src/FeatureContext.cs]: Done! Sorry for the delay.

**Review by halter73 — APPROVED** [review]
Though there seems to be a merge conflict now.

**Comment by paulomorgado** [comment]
Conflicts resolved!

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**
